Dockerfile/base-gcc-clang.Dockerfile

```diff
@@ -23,7 +23,7 @@ RUN    apt-get update \
                 docker.io \
         && apt-get clean \
         && pip install gcovr \
-        && pip install cppyy \
+        && pip install pybind11 \
```

Comment:
I am not sure if it is a good idea to install python packages in the base image since python packages have frequent updates (i.e. pybind11 had 4 releases in PyPI since January). I think the expectation from the base image is to have the least amount of changes, so maybe it would be better to handle the python packages at later stages. Another solution is to pin the version of the python packages as we discussed before, but even in that case base image needs to be updated whenever we change the version.

Reply:
I guess this can be handled later if come to an agreement.

Reply:
@keceli I agree with you. I really don't like how coupled these images are and how far from consumption the dependencies get installed. When one version of anything changes we have to rebuild a ton of images. 

What I have been envisioning is one image per dependency. Docker allows you to compose images, so AFIK that means we basically should be able to mirror how the build system links libraries. For example PluginPlay links to ParallelZone and libfort, so the PluginPlay image should be built on top of the ParallelZone and libfort images. I admittedly don't know if there's a performance hit to having so many images. This also won't prevent all rebuilds, but I think it will limit them. 

That all said, I'm fine prioritizing the switch to Docker in the hopes it speeds up our CI. Once CI is faster I'm hoping it'll be easier to smooth out design.